Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -4030,5 +4065,40 @@ private bool ShouldPreserveDeclaredOutput(AbsolutePath path, HashSet<AbsolutePat
 
             return false;
         }
+
+        private bool IsIncrementalToolAccess(ReportedFileAccess access)
+        {
+            if (!IsIncrementalPreserveOutputPip)
+            {
+                return false;
+            }
+
+            string toolPath = access.Process.Path;
+
+            if (m_incrementalToolFragments.Count == 0)
+            {
+                return false;
+            }
+
+            bool result;
+            if (m_incrementalToolMatchCache.TryGetValue(toolPath, out result))
+            {
+                return result;
+            }
+
+            result = false;
+            foreach (var incrementalToolSuffix in m_incrementalToolFragments)

```

Comment:
since the fragments are relative paths, should we put a path separator in front of them before we check a prefix to avoid:

fragmentname: "foo.exe"
toolPath: "c:\somefoo.exe"

this would be a hit where it shouldn't no?

Reply:
Yes, I already put a path separator in the constructor, please see line #335 in the same class.

Reply:
please see line#335

---
In reply to: [310289947](https://github.com/microsoft/BuildXL/pull/697#discussion_r310289947) [](ancestors = 310289947)